### PR TITLE
mystorm BlackIce support

### PR DESCRIPTION
**Pull request by Fatsie**
Both onboard SRAM and PMOD connectors have been tested.

### CODE REVIEW

File: nmigen_boards/blackice.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+from nmigen.build import *
+from nmigen.vendor.lattice_ice40 import *
+
+
+__all__ = ["BlackIcePlatform"]
+
+
+class BlackIcePlatform(LatticeICE40Platform):
+    device = "iCE40HX8K"
+    package = "TQ144:4K"
```
Comment (whitequark) [inline nmigen_boards/blackice.py]: This doesn't seem right. I think it should be:
```
device = "iCE40HX4K"
package = "TQ144"
```

File: nmigen_boards/blackice.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                dir="io"
+            )),
+            Subsignal("oe", PinsN("45", dir="o")),
+            Subsignal("we", PinsN("120", dir="o")),
+            Subsignal("cs", PinsN("136", dir="o")),
+            Attrs(IO_STANDARD="SB_LVCMOS33"),
+        ),
+    ]
+
+    connectors = [ # TODO: check order of top/bottom row
```
Comment (whitequark) [inline nmigen_boards/blackice.py]: Is this still a TODO? I don't understand what it says anyway.

File: nmigen_boards/blackice.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        Connector("pmod", 0, " 94  91  88  85 - -  95  93  90  87 - -"),  # PMOD1/2
```
Comment (whitequark) [inline nmigen_boards/blackice.py:62]: I wonder if we should use nicer naming than this...

File: nmigen_boards/blackice.py (hunk 1 above)
Comment (whitequark) [reply nmigen_boards/blackice.py]: Ah, I haven't realized the need for the `:4K` suffix. It should not be necessary anymore after m-labs/nmigen@23ed888.

File: nmigen_boards/blackice.py (hunk 2 above)
Comment (Fatsie) [reply nmigen_boards/blackice.py]: No, forgot to remove comment for BlackIce.
Based file on migen boards file and there the double PMOD connectors were split in two. I joined them but made note to check if top row and bottom row matched. As I used the PMOD connectors now the correctness has been verified.
Solved in latest push.

File: nmigen_boards/blackice.py (hunk 1 above)
Comment (Fatsie) [reply nmigen_boards/blackice.py]: Fixed in latest push

**Pull request merged**